### DISCUSSION TITLE
Must gitTimeout be a global option?

### DISCUSSION THREAD

**Discussion by astellingwerf** [renovatebot/renovate#24204] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

v36.67.1

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I'd like to configure the git timeout low in general to prevent useless waiting, and then raise it for specific (known to be large) repositories. I could create a separate pipeline just for the larger repos, but that's extra maintenance I'd prefer to do without.

### What you tried so far.

I had hoped it would work like this:
```
  gitTimeout: 45000,
  packageRules: [
    {
      matchRepositories: ["my/large/repo"],
      gitTimeout: 600000
    }
  ],
```

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
`repositories` in global config can be both a string (repo name) or configuration object, so you could potentially configure it that way in global config as-is. git initialization is done per-repo so it might work
